analysis/org.eclipse.tracecompass.analysis.timing.core.tests/src/org/eclipse/tracecompass/analysis/timing/core/tests/segmentstore/AbstractSegmentStoreStatisticsDataProviderTest.java

```diff
@@ -0,0 +1,208 @@
+         * @param stdDev
+         *            The standard deviation.
+         * @param nbElements
+         *            The number of elements.
+         * @param total
+         *            The total value.
+         * @param minStart
+         *            The minimum start value.
+         * @param minEnd
+         *            The minimum end value.
+         * @param maxStart
+         *            The maximum start value.
+         * @param maxEnd
+         *            The maximum end value.
+         */
+        public StatisticsHolder(String name, long id, long parentId, long min, long max, double average,
+                double stdDev, long nbElements, double total, long minStart, long minEnd, long maxStart, long maxEnd) {
+            fName = name;
+            fId = id;
+            fParentId = parentId;
+            fMin = min;
+            fMax = max;
+            fNbElements = nbElements;
+            fAverage = average;
+            fStdDev = stdDev;
+            fTotal = total;
+            fMinStart = minStart;
+            fMinEnd = minEnd;
+            fMaxStart = maxStart;
+            fMaxEnd = maxEnd;
+        }
+    }
+
+    /**
+     * Holder class for statistics data with an additional user-defined field.
+     */
+    protected static class StatisticsHolderUserDefined extends StatisticsHolder {
+        String fUserDefinedField;
+
+        /**
+         * Constructor for StatisticsHolderUserDefined.
+         *
+         * @param name
+         *            The name of the statistic.
+         * @param id
+         *            The id of the statistic.
+         * @param parentId
+         *            The parent id of the statistic.
+         * @param min
+         *            The minimum value.
+         * @param max
+         *            The maximum value.
+         * @param average
+         *            The average value.
+         * @param stdDev
+         *            The standard deviation.
+         * @param nbElements
+         *            The number of elements.
+         * @param total
+         *            The total value.
+         * @param minStart
+         *            The minimum start value.
+         * @param minEnd
+         *            The minimum end value.
+         * @param maxStart
+         *            The maximum start value.
+         * @param maxEnd
+         *            The maximum end value.
+         * @param userDefinedField
+         *            The user-defined field value.
+         */
+        public StatisticsHolderUserDefined(String name, long id, long parentId, long min, long max, double average,
+                double stdDev, long nbElements, double total, long minStart, long minEnd, long maxStart, long maxEnd, String userDefinedField) {
+            super(name, id, parentId, min, max, average, stdDev, nbElements, total, minStart, minEnd, maxStart, maxEnd);
+            fUserDefinedField = userDefinedField;
+        }
+    }
```

Comment:
Add the following method to reset the static ENTRY_ID counter so that the test cases have known IDs independent of the order of the execution. Call it from in the afterClass method of both test files.

I suggest to use reflection, because I don't want to add an API to reset since it's a common used static member of all segment store data providers. I think it's acceptable to do it like this in test code:

```java
/**
     * Resets static ENTRY_ID of AbstractSegmentStoreStatisticsDataProvider, so that each
     * test class extending AbstractSegmentStoreStatisticsDataProviderTest have a known
     * IDs, independent of the order of execution.
     */
    protected static void resetIds() {
        try {
            Field f = AbstractSegmentStoreStatisticsDataProvider.class.getDeclaredField("ENTRY_ID");
            f.setAccessible(true);
            Object o = f.get(null);
            Class<?> clazz = o.getClass();
            Method m = clazz.getDeclaredMethod("set", long.class);
            m.invoke(o, 0L);
        } catch (NoSuchFieldException | SecurityException |  IllegalArgumentException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
            e.printStackTrace();
        }
    }
```